### PR TITLE
Get-ItemType

### PR DESCRIPTION
**Pull request by SQLvariant**
Pulled the logic for the Get-ItemType "utility" function out of
Write-RsCatalogItem, converted it to a switch statement, and placed it
in the Common folder.

### CODE REVIEW

File: Functions/Common/Get-ItemType.ps1 [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+{
+        param(
+            [string]$FileExtension
+        )
+switch ($FileExtension)
+    {
+        '.rdl'  {return 'Report'}
+        '.rsds' {return 'DataSource'}
+        '.rsd'  {return 'DataSet'}
+        default {throw 'Uploading currently only supports .rdl, .rsds and .rsd files'}
```
Comment (jtarquino) [inline Functions/Common/Get-ItemType.ps1]: Now that is a generic function I would change the exception text to not mention upload

Comment (SQLvariant) [inline Functions/Common/Get-ItemType.ps1]: Ah, good point.

File: Functions/CatalogItems/Write-RsCatalogItem.ps1 [hunk 2]
```diff
@@ -48,116 +63,104 @@ function Write-RsCatalogItem
...
         
+        [Alias('Override')]
         [switch]
-        $Override
+        $OverWrite
     )
-
-    function Get-ItemType
+    
+Begin
```
Comment (jtarquino) [inline Functions/CatalogItems/Write-RsCatalogItem.ps1]: please add indentation

File: Functions/CatalogItems/Write-RsCatalogItem.ps1 [hunk 3]
```diff
@@ -48,116 +63,104 @@ function Write-RsCatalogItem
...
         }
     }
-    
-    if(-not $Proxy)
-    {
-        $Proxy = New-RSWebServiceProxy -ReportServerUri $ReportServerUri -Credentials $ReportServerCredentials 
-    }
 
-    if (!(Test-Path $Path))
+Process
```
Comment (jtarquino) [inline Functions/CatalogItems/Write-RsCatalogItem.ps1]: same here with the indentation

File: Functions/CatalogItems/Write-RsCatalogItem.ps1 [hunk 4]
```diff
@@ -48,116 +63,104 @@ function Write-RsCatalogItem
...
     {
-        throw "No item found at the specified path: $Path!"
-    }
+    foreach ($item in $Path)
```
Comment (jtarquino) [inline Functions/CatalogItems/Write-RsCatalogItem.ps1]: indentation

**Pull request merged**